docs/4-Prototyping.md

```diff
@@ -1,29 +1,172 @@
 # Prototyping
-## Overview
+
+A website prototype can be any mock-up or demo of what a website could look like when it goes live. A living prototype is a prototype that's actively used throughout the lifecycle of the website. Even after the site is deployed, the prototype lives on as an active demonstration of what the website looks like.
+
+The following features and conventions come together to allow the creation of a living prototype.
+
+## Conventions
+
+### Meta-Data
+
+Meta-Data is general data that helps describe the content you're presenting. Depending on the scope, meta-data can exist almost anywhere.
+
+#### Local
+
+Add YAML front-matter to files to append properties to the file.
+
+*example.html*
+``` html
+---
+name: Linus Torvalds
+languages:
+- PHP
+- Perl
+- Python
+---
+<h1>Linus Torvalds</h1>
+```
+
+*output*
+``` html
+<h1>Linux Torvalds</h1>
+```
+
+#### Global
+
+You can add global meta-data through the use of the [metalsmith-metadata-convention](https://www.npmjs.com/package/metalsmith-metadata-convention) plugin and `.metadata` files.
+
+*authors.metadata*
+``` yaml
+---
+Stephen King:
+  birthdate: 1947
+J. K. Rowling:
+  birthdate: 1965
+William Shakespeare:
+  birthdate: 1564
+Nora Roberts:
+  birthdate: 1950
+---
+
+This list of authors is loaded into a global metadata variable named "authors".
+```
+
+### Transformers
+
+Naming a file by a [JSTransformer](https://github.com/jstransformers/jstransformer) name will have it process with the engine. The following is an example of using [Twig](http://twig.sensiolabs.org):
+
+*example.html.twig*
+``` twig
+---
+name: Linus Torvalds
+---
+<h1>{{ name }}</h1>
+```
+
+*output*
+``` html
+<h1>Linus Torvalds</h1>
+```
+
+See [JSTransformers](https://github.com/jstransformers) for a list of available transformers, and [metalsmith-jstransformer](https://www.npmjs.com/package/metalsmith-jstransformer) to see what's possible. The following is a list of transformers that are available through Kalastatic:
+
+- [pug](http://npm.im/jstransformer-pug)
+- [twig](http://npm.im/jstransformer-twig)
+- [browserify](http://npm.im/jstransformer-browserify)
+- [scss](http://npm.im/jstransformer-scss)
+- [md](http://npm.im/jstransformer-commonmark)
+
+### Passing in environment variables
+### Meta-Data Files
+
+The [metalsmith-metadata-files](https://github.com/kalamuna/metalsmith-metadata-files) plugin will inject sister JSON files into the main file.
+
+*example.html.json*
+``` json
+{
+  "name": "Linus Torvalds"
+}
+```
+
+*example.html.twig*
+``` twig
+<h1>{{ name }}</h1>
+```
+
+*output*
+```
+<h1>Linux Torvalds</h1>
+```
+
+### Globals
+- kalastatic.yml
+
+### Assets
+
+The [metalsmith-assets-convention](https://github.com/robloach/metalsmith-assets-convention) plugin will deploy any local assets you would like to copy out through the use of `.assets` files:
```

Comment:
Maybe expand a little on why you need to do this? e.g. so you can pull in a js library via npm and then use it in your site.